tls-tutorial/doc/use_your_own_certificate.md

```diff
@@ -0,0 +1,736 @@
+   ```shell
+   sudo mysql -u root
+   ```
+3. Create a database `shapes` and a user `tutorial_user` for our import experiment
+   ```sql
+   create database shapes;
+   create user 'tutorial_user' identified by 'tutorial';
+   grant all on shapes.* to tutorial_user;
+   exit
+   ```
+4. Login as a `tutorial_user`
+   ```sql
+   mysql -u tutorial_user -p
+   ```
+5. Populate the database
+   ```sql
+   create table shapes.shapes (name VARCHAR(40), corners int);
+   insert into shapes.shapes values ('point', 1), ('line', 2), ('triangle', 3), ('rectangle', 4);
+   exit
+   ```
+
+As you can tell we won't win any price for most original database usage here, but that is not the point of the tutorial.
+
+### Installing Certificates in MySQL
+
+By default, the MySQL server on Ubuntu comes with a self-signed certificate. Since self-signed certificates don't originate from a CA the client knows, clients will be prompted to trust them.
+
+This is a bad practice, and since we already went through the trouble of creating our own certificates, our next job will be to use them with MySQL.
+
+1. Install the CA certificate centrally:
+   ```shell
+   sudo cp ca.crt /usr/local/share/ca-certificates/exasol_tutorial_ca.crt
+   sudo update-ca-certificates
+   ```
+2. Verify that it was installed:
+   ```shell
+   ls -l /etc/ssl/certs/exasol*
+   ```
+   Must output something like:
+   ```
+   lrwxrwxrwx 1 root root 55 Jan 12 11:43 /etc/ssl/certs/exasol_tutorial_ca.pem -> /usr/local/share/ca-certificates/exasol_tutorial_ca.crt
+   ```
+   Note how the symbolic link changes the file suffice from `.crt` to `.pem`
+3. Install the server certificate and key
+   ```shell
+   sudo mkdir -p /etc/mysql/certs
+   sudo cp server.crt /etc/mysql/certs/
+   sudo cp server.key /etc/mysql/certs/
+   sudo chown mysql:mysql /etc/mysql/certs/server.key
+   ```
+
+Note that we need `sudo` privileges here, since we need to install the certificates in a central locations that belong to the `root` user.
+
+Additionally, we give the private key to the MySQL user on OS level. This user is aptly called `mysql`, belongs to a group of the same name and is created automatically when you install MySQL on Ubuntu. We don't want anyone else to be able to read it, so weakening permissions on that file is not an option.
+
+You can check ownership and permissions like this:
+
+```shell
+ls -l /etc/mysql/certs/
+```
+
+This should list two files:
+
+```
+-rw-r--r-- 1 root  root  1919 Jan 12 11:56 server.crt
+-rw------- 1 mysql mysql 3326 Jan 12 11:56 server.key
+```
+
+Now edit the `[mysqld]` section of the MySQL configuration file `/etc/mysql/mysql.conf.d/mysqld.cnf `as `root` user and add:
+
+```
+[mysqld]
+ssl
+ssl-ca=/etc/ssl/certs/exasol_tutorial_ca.pem
+ssl-cert=/etc/mysql/certs/server.crt
+ssl-key=/etc/mysql/certs/server.key
+```
+
+Also, we need to enable Exasol later to access the MySQL server. Since that will not be access from `localhost` you need to accept all interfaces.
```

Comment:
```suggestion
Also, we need to enable Exasol later to access the MySQL server. Since that access will not be from `localhost`, you need to accept all interfaces.
```